.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,19 @@
+---
+name: Pre Commit
+on:
+  pull_request:
+
+
+jobs:
+  pre-commit:
+    name: Pre Commit Hooks
+    runs-on: ubuntu-latest
+    steps:
+    - run: sudo apt-get install -y python3-rpm
+    - uses: actions/checkout@v3
+      with:
+        fetch-depth: 0
+    - uses: actions/setup-python@v3
+      with:
+        python-version: '3.10'
+    - uses: pre-commit/action@v3.0.0
```

Comment:
I would suggest this instead:

```
- name: Install pre-commit
  run: pip install pre-commit
- name: Run pre-commit on entire codebase
  run: pre-commit run -a
```
It does look a bit clunky, but it would remove dependency on an unmaintained action.

https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun

Reply:
I like this, I fixed it.
I also removed clang now because the output is quite similar and fixed the pre commit issues. I believe it's good to go now. @m-blaha